contrib/colab/PyTorch_TPU_XRT_1_13.ipynb

```diff
@@ -0,0 +1,563 @@
+        "    return F.log_softmax(x, dim=1)\n",
+        "\n",
+        "print (\"mnist defined\")\n",
+        "def train_mnist():\n",
+        "  assert FLAGS.num_cores == 1\n",
+        "  torch.manual_seed(1)\n",
+        "  # Training settings\n",
+        "  lr = 0.01\n",
+        "  momentum = 0.5\n",
+        "  log_interval = 5\n",
+        "\n",
+        "  if FLAGS.fake_data:\n",
+        "    print('using fake data')\n",
+        "    train_loader = xu.SampleGenerator(\n",
+        "        data=torch.zeros(FLAGS.batch_size, 1, 28, 28),\n",
+        "        target=torch.zeros(FLAGS.batch_size, dtype=torch.int64),\n",
+        "        sample_count=60000 // FLAGS.batch_size)\n",
+        "    test_loader = xu.SampleGenerator(\n",
+        "        data=torch.zeros(FLAGS.batch_size, 1, 28, 28),\n",
+        "        target=torch.zeros(FLAGS.batch_size, dtype=torch.int64),\n",
+        "        sample_count=10000 // FLAGS.batch_size)\n",
+        "  else:\n",
+        "    print('using real data', FLAGS.datadir)\n",
+        "    train_loader = torch.utils.data.DataLoader(\n",
+        "        datasets.MNIST(\n",
+        "            FLAGS.datadir,\n",
+        "            train=True,\n",
+        "            download=True,\n",
+        "            transform=transforms.Compose([\n",
+        "                transforms.ToTensor(),\n",
+        "                transforms.Normalize((0.1307,), (0.3081,))\n",
+        "            ])),\n",
+        "        batch_size=FLAGS.batch_size,\n",
+        "        shuffle=True,\n",
+        "        num_workers=FLAGS.num_workers)\n",
+        "    test_loader = torch.utils.data.DataLoader(\n",
+        "        datasets.MNIST(\n",
+        "            FLAGS.datadir,\n",
+        "            train=False,\n",
+        "            transform=transforms.Compose([\n",
+        "                transforms.ToTensor(),\n",
+        "                transforms.Normalize((0.1307,), (0.3081,))\n",
+        "            ])),\n",
+        "        batch_size=FLAGS.batch_size,\n",
+        "        shuffle=True,\n",
+        "        num_workers=FLAGS.num_workers)\n",
+        "\n",
+        "  model = MNIST()\n",
+        "\n",
+        "  inputs = torch.zeros(FLAGS.batch_size, 1, 28, 28)\n",
+        "  xla_model = xm.XlaModel(model, [inputs])\n",
+        "  optimizer = optim.SGD(xla_model.parameters_list(), lr=lr, momentum=momentum)\n",
+        "  loss_fn = nn.NLLLoss()\n",
+        "  accuracy = None\n",
+        "  for epoch in range(1, FLAGS.num_epochs + 1):\n",
+        "    # Training loop for epoch.\n",
+        "    start_time = time.time()\n",
+        "    processed = 0\n",
+        "    for batch_idx, (data, target) in enumerate(train_loader):\n",
+        "      if data.size()[0] != FLAGS.batch_size:\n",
+        "        break\n",
+        "      optimizer.zero_grad()\n",
+        "      y = xla_model(data)\n",
+        "      y[0].requires_grad = True\n",
```

Comment:
I am not sure this line is needed. Can you try to remove and check?

Reply:
sorry- which line did you want me to remove?

Reply:
The requires_grad one.
Can you try without?